src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -1976,14 +1863,21 @@ private byte[] generateNamedFunctionInvokerImpl(MethodTypeForm typeForm) {
      * Emit a bogus method that just loads some string constants. This is to get the constants into the constant pool
      * for debugging purposes.
      */
-    private void bogusMethod(Object os) {
+    private void bogusMethod(ClassBuilder clb, Object os) {
         if (DUMP_CLASS_FILES) {
-            mv = cw.visitMethod(Opcodes.ACC_STATIC, "dummy", "()V", null, null);
-            mv.visitLdcInsn(os.toString());
-            mv.visitInsn(Opcodes.POP);
-            mv.visitInsn(Opcodes.RETURN);
-            mv.visitMaxs(0, 0);
-            mv.visitEnd();
+            clb.withMethod("dummy", MethodTypeDesc.of(CD_void), ACC_STATIC, new Consumer<MethodBuilder>() {
+                @Override
+                public void accept(MethodBuilder mb) {
+                    mb.withFlags(ACC_STATIC);
+                    mb.withCode(new Consumer<CodeBuilder>() {
+                        @Override
+                        public void accept(CodeBuilder cob) {
+                            cob.constantInstruction(os.toString());
+                            cob.pop();
+                            cob.return_();
+                        }
+                    }); }
+            });
```

Comment:
I suggest using `withMethodBody` here as well, since the separate call to `withFlags` is not needed.

Reply:
Fixed, thanks.